src/main/java/it/pagopa/transactions/configurations/AzureStorageConfig.java

```diff
@@ -66,20 +87,59 @@ public QueueAsyncClient transactionNotificationRequestedQueueAsyncClient(
                                                                              ) String storageConnectionString,
                                                                              @Value(
                                                                                  "${azurestorage.queues.transactionnotificationrequested.name}"
-                                                                             ) String queueName
+                                                                             ) String queueName,
+                                                                             JsonSerializer jsonSerializer
     ) {
-        return buildQueueAsyncClient(storageConnectionString, queueName);
+        return buildQueueAsyncClient(storageConnectionString, queueName, jsonSerializer);
     }
 
     private QueueAsyncClient buildQueueAsyncClient(
                                                    String storageConnectionString,
-                                                   String queueName
+                                                   String queueName,
+                                                   JsonSerializer jsonSerializer
     ) {
-        QueueAsyncClient queueAsyncClient = new QueueClientBuilder()
+        com.azure.storage.queue.QueueAsyncClient queueAsyncClient = new QueueClientBuilder()
                 .connectionString(storageConnectionString)
                 .queueName(queueName)
+                .addPolicy(
+                        (
+                         context,
+                         next
+                        ) -> {
+                            HttpHeaders requestHeaders = context.getHttpRequest().getHeaders();
+
+                            Set<String> rebrandedHeaders = Set
+                                    .of("traceparent", "tracestate", "baggage");
+
+                            for (String header : rebrandedHeaders) {
+                                context.getHttpRequest().setHeader("pagopa-" + header, requestHeaders.getValue(header));
+                            }
+
+                            context.getHttpRequest().setHeader("pagopa-foo", "bar");
```

Comment:
To remove?

Reply:
Yes, I will remove this (and draft) once all dependant PRs are merged :)